### PR TITLE
Internal: Plugins Tester - Fixes [ED-9673]

### PR DESCRIPTION
**Pull request by matipojo**
- Updated list
- Create artifacts only on failure
- Include only failed plugins in the artifact

### CODE REVIEW

File: .github/workflows/plugins-tester.yml [hunk 1]
```diff
@@ -83,14 +83,8 @@ jobs:
       - name: Run tests
```
Comment (StyleShit) [inline .github/workflows/plugins-tester.yml:83]: why did you remove the package lock?

File: tests/plugins-tester/config/plugins.js [hunk 2]
```diff
@@ -2,10 +2,14 @@
 import topPluginsConfig from "./top-plugins.json" assert {type: 'json'};
 
 const notCompatiblePlugins = [
-	'essential-addons-for-elementor-lite',
+	'complianz-gdpr', // SQL Error
+	'easy-table-of-contents', // Welcome page.
```
Comment (StyleShit) [inline tests/plugins-tester/config/plugins.js:6]: what does it mean? (also the other one)

File: tests/plugins-tester/package.json [hunk 3]
```diff
@@ -6,11 +6,11 @@
   "type": "module",
   "scripts": {
     "wp-env": "wp-env",
-    "test": "node index.js"
+    "test": "node index.js",
+	"update-baseline": "node ./scripts/run-backstop.js --slug='elementor'"
```
Comment (StyleShit) [inline tests/plugins-tester/package.json:10]:
```suggestion
    "test": "node index.js",
    "update-baseline": "node ./scripts/run-backstop.js --slug='elementor'"
```

File: tests/plugins-tester/scripts/update-top-plugins-list.js [hunk 4]
```diff
@@ -22,6 +22,9 @@ const fetchPluginsList = async ( url, filename ) => {
             pluginsList.push( plugin.slug );
         } );
 
+		// Sort the list
+		pluginsList.sort();
+
```
Comment (StyleShit) [inline tests/plugins-tester/scripts/update-top-plugins-list.js:27]: lint 
(seems like the file uses spaces)

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: tests/plugins-tester/scripts/update-top-plugins-list.js (hunk 4 above)
Comment (matipojo) [reply tests/plugins-tester/scripts/update-top-plugins-list.js]: Done

File: .github/workflows/plugins-tester.yml (hunk 1 above)
Comment (StyleShit) [reply .github/workflows/plugins-tester.yml:83]: Oh, oops

File: tests/plugins-tester/scripts/update-top-plugins-list.js [hunk 5]
```diff
@@ -2,37 +2,41 @@ import fetch from 'node-fetch';
...
-        data.plugins.forEach( ( plugin ) => {
-            if ( 'elementor' === plugin.slug ) {
-                return;
-            }
+			pluginsList.push( plugin.slug );
+		} );
 
-            pluginsList.push( plugin.slug );
-        } );
+		// Sort the list
```
Comment (StyleShit) [inline tests/plugins-tester/scripts/update-top-plugins-list.js:26]:
```suggestion
``` 
I think the code is self-explanatory

**Review by StyleShit — CHANGES_REQUESTED** [review]

**Review by StyleShit — APPROVED** [review]
